drivers/mpu9250/MPU9250.cpp

```diff
@@ -160,8 +160,15 @@ int MPU9250::mpu9250_init()
 	//	DF_LOG_ERR("sample rate config failed");
 	//}
 	//usleep(1000);
+
+#if defined(__EDISON)
+	//FIFO corrupt and overflow at 250Hz.
+	result = _writeReg(MPUREG_CONFIG,
+			   BITS_DLPF_CFG_184HZ | BITS_CONFIG_FIFO_MODE_OVERWRITE);
```

Comment:
That's why it used the FIFO buffer of the device. The current driver is actually running at 1000 Hz (see: https://github.com/PX4/DriverFramework/blob/master/drivers/mpu9250/MPU9250.hpp#L182), so it collects 8 samples each time. 

There were some problems when the mag was added but they have recently been fixed, see #117.